packages/test/test-service-load/src/leaderElection.ts

```diff
@@ -0,0 +1,111 @@
+        this.dataStoreRuntime = dataStoreRuntime;
+        this.clientId = clientId;
+        this.logger = logger;
+    }
+
+    public setupLeaderElection() {
+        this.electNewLeader("Initializing leader election.");
+        this.dataStoreRuntime.on("signal", (signal: ISignalMessage) => this.handleSignal(signal));
+        this.dataStoreRuntime.getAudience().on("removeMember", (clientId) => {
+            if(clientId === this.leaderId) {
+                this.electNewLeader("Leader was removed from the audience.");
+            }
+        });
+
+        const interval = setInterval(() => {
+            if (this.leaderId !== undefined && this.leaderId === this.clientId) {
+                this.dataStoreRuntime.submitSignal("leaderMessage", "leaderMessage");
+                this.lastPinged = new Date();
+                console.log(`Leader with client id: ${this.leaderId} sending ping at ${Date.now()}.`);
+                this.logger.send(
+                    {
+                        category: "performance",
+                        eventName: "LeaderElection:PingSent",
+                        leaderId: this.leaderId,
+                        time: Date.now(),
+                    });
+            }else if(this.leaderId === undefined) {
+                this.electNewLeader("Leader is undefined.");
+            }else {
+                const current = new Date();
+                if(this.lastPinged === undefined || current.getTime() - this.lastPinged?.getTime() > this.leaderWait) {
+                    this.electNewLeader("Not recieving leader messages.");
+                }
+            }
+        }, this.beatInEveryNSecs);
+
+        this.dataStoreRuntime.once("dispose", () => {
+            clearInterval(interval);
+        });
+    }
+
+    private handleSignal(signal: ISignalMessage) {
+        // eslint-disable-next-line no-null/no-null
+        if(signal.clientId !== null && signal.content === "leaderMessage") {
+            if(this.leaderId === signal.clientId) {
+                console.log(`Expected leader message recieved: ${this.leaderId} 
+                at client ${this.clientId}`);
+            }else{
+                console.log(`Client id: ${this.clientId}
+                recieving ping at ${Date.now()}
+                from ${signal.clientId}
+                expecting leader ${this.leaderId}.`);
+                this.logger.send(
+                    {
+                        category: "performance",
+                        eventName: "LeaderElection:ElectNewLeader",
+                        reason: "Recieved unexpected leader id.",
+                        clientId: this.clientId,
+                        newLeaderId: signal.clientId,
+                        oldLeaderId: this.leaderId,
+                    });
+                this.leaderId = signal.clientId;
+            }
+            this.lastPinged = new Date();
+        }
+    }
+
+    private electNewLeader(reason: string) {
```

Comment:
i think leader election can be simplified. i would just do it all in a property:
```
    private get leaderId(){return Array.from(this.dataStoreRuntime.getAudience().getMembers().keys()).sort()[0];}
```

Reply:
i like the instinct to log, but i think this is overly verbose. i really only want to see a log entry when the signal come from a client that is not the leader, and i want that to log an error